library/std/src/os/unix/net/ancillary.rs

```diff
@@ -441,10 +481,24 @@ impl<'a> SocketAncillary<'a> {
         self.length
     }
 
+    /// Set the number of valid ancillary data bytes and the truncated flag.
+    ///
+    /// This can be used with [`buffer_mut()`](Self::buffer_mut)
+    /// to manually write ancillary data into the buffer.
+    ///
+    /// # Safety
+    /// - The length may not exceed [`capacity()`](Self::capacity).
+    /// - The data in the buffer at `0..length` must be valid ancillary data.
+    #[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+    pub unsafe fn set_len(&mut self, length: usize, truncated: bool) {
```

Comment:
I could not think of situations where you want to change `length` and the `truncated` flag independently of each-other, so I figured it makes sense to ask for both at the same time. That way you can't accidentally forget to update the truncated flag.